pallets/did/src/tests.rs

```diff
@@ -1670,6 +1679,81 @@ fn check_too_large_tx_counter_call_error() {
 	});
 }
 
+#[test]
+fn check_tx_block_number_too_low_error() {
+	let auth_key = get_sr25519_authentication_key(true);
+	let did = get_did_identifier_from_sr25519_key(auth_key.public());
+	let caller = ACCOUNT_00;
+
+	let mock_did = generate_base_did_details::<Test>(did::DidVerificationKey::from(auth_key.public()));
+
+	let mut ext = ExtBuilder::default()
+		.with_dids(vec![(did.clone(), mock_did)])
+		.build(None);
+
+	let submitter = kilt_primitives::AccountId::default();
+
+	let call_operation = generate_test_did_call(did::DidVerificationKeyRelationship::Authentication, did, submitter);
+	let signature = auth_key.sign(call_operation.encode().as_ref());
+
+	ext.execute_with(|| {
+		// System block number 1 past the max block the operation was allowed for.
+		System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get() + 1);
+		assert_noop!(
+			Did::submit_did_call(
+				Origin::signed(caller.clone()),
+				Box::new(call_operation.operation.clone()),
+				did::DidSignature::from(signature.clone())
+			),
+			did::Error::<Test>::InvalidOperationValidity
+		);
+	});
+
+	// But it would work if the system would be one block earlier.
+	ext.execute_with(|| {
+		System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get());
+		assert_ok!(Did::submit_did_call(
+			Origin::signed(caller),
+			Box::new(call_operation.operation),
+			did::DidSignature::from(signature)
+		),);
+	});
```

Comment:
W: See above regarding multiple Externalities.

```suggestion
	ExtBuilder::default()
		.with_dids(vec![(did.clone(), mock_did)])
		.build(None).
		execute_with(|| {
			let submitter = kilt_primitives::AccountId::default();
			let call_operation = generate_test_did_call(did::DidVerificationKeyRelationship::Authentication, did, submitter);
			let signature = auth_key.sign(call_operation.encode().as_ref());
					
			// System block number 1 past the max block the operation was allowed for.
			System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get() + 1);
			assert_noop!(
				Did::submit_did_call(
					Origin::signed(caller.clone()),
					Box::new(call_operation.operation.clone()),
					did::DidSignature::from(signature.clone())
				),
				did::Error::<Test>::InvalidOperationValidity
			);

			// But it would work if the system would be one block earlier.
			System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get());
			assert_ok!(Did::submit_did_call(
				Origin::signed(caller),
				Box::new(call_operation.operation),
				did::DidSignature::from(signature)
			),);
		});
```

Reply:
I guess as you said it will be the scope of a different PR in the future

Reply:
Yes.

Reply:
Since you touched this test, i would already apply Williams suggestion.

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/257/commits/639389dc8c1f3317b4a523fa6225412a21d53b01